### PR TITLE
Add ERROR_FILE_EXISTS to ErrorKind conversion on Windows

### PR DESCRIPTION
**Pull request by gkoz**
Bug report: https://users.rust-lang.org/t/detecting-error-kind-for-opening-file/6215
Reference: https://msdn.microsoft.com/en-us/library/windows/desktop/ms681382(v=vs.85).aspx#error_file_exists

### CODE REVIEW

**Comment by Stebalien** [comment]
How hard would it be to add a platform-independent test case for this (so it doesn't show up again)?

**Comment by gkoz** [comment]
Sorry, it's not obvious to me whether by 'this' you mean inaccurate `ErrorKind::Other` in general or something more specific.

**Comment by alexcrichton** [comment]
I believe what @Stebalien means is could we add a test for this? Ideally one that runs on both Unix and Windows and both situations give us `AlreadyExists`

**Comment by Stebalien** [comment]
> I believe what @Stebalien means is could we add a test for this? Ideally one that runs on both Unix and Windows and both situations give us AlreadyExists

Ideally one that doesn't check for any specific platforms.

**Comment by gkoz** [comment]
Added a test.

**Comment by alexcrichton** [comment]
@bors: r+ 973ec7cfa9a14bc825102188e8ad40ff3d733cda 

Thanks!

File: src/libstd/fs.rs [hunk 1]
```diff
@@ -1771,12 +1771,29 @@ mod tests {
...
     #[test]
+    fn file_create_new_path_already_exists_error() {
+        let tmpdir = tmpdir();
+        let file = &tmpdir.join("file_create_new_exists_file");
+        check!(fs::File::create(file));
+        let e = fs::OpenOptions::new().write(true).create_new(true).open(file).unwrap_err();
+        assert_eq!(e.kind(), ErrorKind::AlreadyExists);
+        let dir = &tmpdir.join("file_create_new_exists_dir");
+        check!(fs::create_dir(dir));
+        let e = fs::OpenOptions::new().write(true).create_new(true).open(dir).unwrap_err();
```
Comment (gkoz) [inline src/libstd/fs.rs]: Apparently on Windows an this (creating a file when e directory exists) results in `ErrorKind::PermissionDenied`. Should I just not test this or make it conditional on `!windows`?

Comment (alexcrichton) [inline src/libstd/fs.rs]: It's fine to just omit this test if the PR is covered by the one above.

**Comment by gkoz** [comment]
Removed fancy test cases.

**Pull request merged**